scripts/rogue-keeper-upper.php

```diff
@@ -0,0 +1,300 @@
+
+        // Put request in failed table for future investigation
+        // @TODO: only put in this table if it's not already there
+        dosomething_rogue_handle_migration_failure($data, $post->sid, $post->rbid, $post->fid, $response, $e);
+      }
+      catch (DoSomething\Gateway\Exceptions\ApiException $e) {
+        echo 'api exception' . PHP_EOL;
+        // Put request in failed table for future investigation
+        dosomething_rogue_handle_migration_failure($data, $post->sid, $post->rbid, $post->fid, $response, $e);
+      }
+  }
+  else {
+    echo 'No northstar id, that is terrible ' . $post->fid . PHP_EOL;
+
+    // Put request in failed table for future investigation
+    dosomething_rogue_handle_migration_failure($data, $post->sid, $post->rbid, $post->fid);
+  }
+}
+
+// 3. Send all new posts
+// Get all posts that are not in Rogue, but belong to a signup that IS in Rogue
+// Do not include posts that are already in the failed table
+$posts = db_query('SELECT rbf.fid, rb.rbid, rb.flagged, rbf.source, rbf.remote_addr, rbf.caption, rbf.status, rb.nid, rb.run_nid, signup.sid, rblog.timestamp, rb.uid, rb.nid, rb.run_nid, rb.quantity, rb.why_participated
+                  FROM dosomething_reportback_file rbf
+                  LEFT JOIN dosomething_reportback rb ON rbf.rbid = rb.rbid
+                  JOIN dosomething_signup signup ON signup.uid = rb.uid
+                    AND signup.nid = rb.nid
+                    AND signup.run_nid = rb.run_nid
+                  JOIN dosomething_reportback_log rblog ON rbf.fid = substring_index(rblog.files, \',\',-1)
+                  WHERE rbf.fid NOT IN (SELECT fid FROM dosomething_rogue_reportbacks)
+                    AND rbf.fid NOT IN (SELECT fid FROM dosomething_rogue_failed_migrations WHERE fid IS NOT NULL)
+                    AND signup.sid IN (SELECT sid from dosomething_rogue_signups)
+                  GROUP BY rbf.fid');
+
+// Prepare client to send to Rogue
+$client = dosomething_rogue_client();
+
+// Iterate through all the new posts and send them off
+foreach ($posts as $post) {
+  $data = [];
+
+  echo 'Trying fid ' . $post->fid . ' sid ' . $post->sid . '...' . PHP_EOL;
+
+  // Get the user info
+  $northstar_user = dosomething_northstar_get_user($post->uid, 'drupal_id');
+
+  if (isset($northstar_user)) {
+  // Format the Post data to send to Rogue
+    // Get the status as a Rogue status
+    $rogue_status = dosomething_rogue_transform_status($post->status);
+
+    // Match Rogue's timestamp format
+    $photo_created_at = date('Y-m-d H:i:s', $post->timestamp);
+
+    // Format the photo data
+    $data = [
+      // Post
+      'northstar_id' => $northstar_user->id,
+      'campaign_id' => $post->nid,
+      'campaign_run_id' => $post->run_nid,
+      'caption' => $post->caption,
+      'source' => $post->source,
+      'remote_addr' => $post->remote_addr,
+      'status' => $rogue_status, //can you send this? not in docs
+      'do_not_forward' => TRUE,
```

Comment:
@DFurnes we don't use it, but in some cases I was getting errors without this because Rogue was still trying to. Removing that from the Rogue side is for sure the better solution since we don't ever want to do it now, I just made an issue for it so it doesn't get forgotten https://github.com/DoSomething/rogue/issues/241